index.rst

```diff
@@ -1,51 +1,1166 @@
+
+.. code-block:: c++
+
+    #include <pybind11/pybind11.h>
+    //#include <pybind11/operators.h>
+    #include <pybind11/stl.h>
+
+    #include "lsst/afw/math/minimize.h"
+
+    namespace py = pybind11;
+
+    using namespace lsst::afw::math;
+
+    PYBIND11_PLUGIN(_minimize) {
+        py::module mod("_minimize", "Python wrapper for afw _minimize library");
+
+        py::class_<FitResults> clsFitResults(mod, "FitResults");
+
+        clsFitResults.def_readwrite("isValid", &FitResults::isValid);
+        clsFitResults.def_readwrite("chiSq", &FitResults::chiSq);
+        clsFitResults.def_readwrite("parameterList", &FitResults::parameterList);
+        clsFitResults.def_readwrite("parameterErrorList", &FitResults::parameterErrorList);
+
+        return mod.ptr();
+    }
+
+This is a good time to build our changes (at times the error messages generated by pybind11 can be obscure so it is useful to recompile after each wrapped class).
+From the shell prompt run
+
+.. code-block:: bash
+
+    scons lib python
+
+to build all of the changes you made to afw.
+If the build failed, go back and verify that all of your function definitions used the correct syntax as displayed above.
+
+Wrapping an overloaded function
+-------------------------------
+
+Now that we have created the ``FitResults`` structure we can create our ``minimize`` function wrapper.
+This is done using the ``def`` method of ``py::module``, where we must create a definition for each set of parameters.
+Looking in the swig ``.i`` file located at https://github.com/lsst/afw/blob/master/python/lsst/afw/math/minimize.i we see that there are two templated types: ``float`` and ``double``.
+
+.. note::
+
+    Whenever you encounter a problem that requires you to look at the swig files you are best off looking at the code on github, as the swig files have been deleted in the pybind11 branch and switching branches locally will require you to commit or stash your changes, which might be inconvenient at the time.
+
+In a minute we will wrap ``minimize`` for both types, but it is useful to first look at how this would be done for a single type ``double``.
+In this case we define ``minimize`` and cast it to a ``FitResults`` function pointer underneath our ``clsFitResults`` code using
+
+.. code-block:: c++
+
+    mod.def("minimize", (FitResults (*) (lsst::afw::math::Function1<double> const &,
+                                         std::vector<double> const &,
+                                         std::vector<double> const &,
+                                         std::vector<double> const &,
+                                         std::vector<double> const &,
+                                         std::vector<double> const &,
+                                         double)) minimize<double>);
+
+Notice that for each parameter in the C++ function we include the type (including a reference if necessary) in our pybind11 function declaration but not the variable name itself.
```

Comment:
Good point, that is not the case. The main point I was trying to make here is to remove the variable name.